Project1/Tile.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "Terrain.h"
+#include "CellEdge.h"
+#include <vector>
+#include "Unit.h"
+/** Tile held within a GridCell. Has no position information to keep the implementation OO
+*/
+class Tile {
+private:
+	
+	Terrain* _terrain; //Not sure how i want to do terrain yet, this is placeholder i think it should be some kind of pointer to some static object.
+	Unit* _unit = nullptr;
+public:
+	Tile();
+	~Tile();
+	bool insertUnit(Unit* unit);
```

Comment:
If this is going to have getters and setters for both, might as well make them public members.
I'd make Unit* public and make it so the Terrain is set by the constructor and cannot be changed.